.github/workflows/unit_test_report.yaml

```diff
@@ -0,0 +1,101 @@
+name: Upload Unit Test Results
+
+on:
+  workflow_run:
+    workflows:
+      - evergreen
+      - linux
+    types:
+      - completed
+
+jobs:
+  unit-test-report:
+    permissions: {}
+      # TODO(b/293508740): Report failed workflow runs back to triggering PR.
+      # pull-requests: write
+    if: always()
+    runs-on: ubuntu-latest
+    name: Upload Unit Test Reports
+    steps:
+      - name: Download Unit Test Results
+        # The `download-artifact` action can only access artifacts that were uploaded in the same workflow.
+        # Since it was not this workflow that uploaded the artifacts we must use rest api to download them.
+        # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#using-data-from-the-triggering-workflow
+        uses: actions/github-script@v6
+        with:
+          script: |
+            const name = context.payload.workflow_run.name;
+            const url = context.payload.workflow_run.html_url;
+            console.log(`Downloading unit test results for '${name}' (${url}).`);
+
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              run_id: context.payload.workflow_run.id
+            });
+
+            let matchArtifacts = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "unit-test-results"
+            });
+
+            if (matchArtifacts.length == 1) {
+              let download = await github.rest.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: matchArtifacts[0].id,
+                archive_format: 'zip'
+              });
+              let fs = require('fs');
+              const path = `${process.env.GITHUB_WORKSPACE}/unit-test-results.zip`;
+              fs.writeFileSync(path, Buffer.from(download.data));
+            } else {
+              core.setFailed(`Expected one artifact with name 'unit-test-results'. Found ${matchArtifacts.length}.`);
+            }
+      - name: Extract Archived Unit Test Results
+        run: unzip unit-test-results.zip -d unit-test-results
+      - name: Install node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: Get Datadog CLI
+        shell: bash
+        # TODO: pin version (with checksum?)
+        run: npm install -g @datadog/datadog-ci
```

Comment:
We don't have a repo here which means no packages.json file to pin the version and package hashes.

`datadog-ci` is also provided as a [standalone executable](https://github.com/DataDog/datadog-ci#more-ways-to-install-the-cli). I could download a specific version and verify the hash. They are in beta and are provided as-is but I think it's probably still safe if we download a specific version.

Reply:
Currently doing this on https://github.com/youtube/cobalt_sandbox/commit/03fa19c5b877ddf8953e76088791491053a3fb34

Reply:
This worked like a charm.